src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java

```diff
@@ -35,77 +42,176 @@
-    public void secretFileBaseTest() throws IOException, URISyntaxException {
+    public void secretFileGlobalScopeTest() throws IOException, URISyntaxException {
         DiskFileItem fileItem = createEmptyFileItem();
-        
-        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret file", fileItem, "keys.txt", Base64.encode(fileItem.get()).toString());
+
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret file", fileItem, "keys.txt", SecretBytes.fromBytes(fileItem.get()));
         FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getSecretBytes());
-        testCreateUpdateDelete(credential, updatedCredential);
+        testCreateUpdateDelete(credential, updatedCredential, null);
+    }
+
+    @Test
+    public void secretFileSystemScopeTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.SYSTEM, CRED_ID, "Test Secret file", fileItem, "keys.txt", SecretBytes.fromBytes(fileItem.get()));
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getSecretBytes());
+        testCreateUpdateDelete(credential, updatedCredential, null);
+    }
+
+    @Test
+    public void secretFilePersonalScopeTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.USER, CRED_ID, "Test Secret file", fileItem, "keys.txt", SecretBytes.fromBytes(fileItem.get()));
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getSecretBytes());
+        testCreateUpdateDelete(credential, updatedCredential, null);
+    }
+
+    @Test
+    public void secretFileDomainTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret file", fileItem, "keys.txt", SecretBytes.fromBytes(fileItem.get()));
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getSecretBytes());
+
+        DomainSpecification[] ar = { new HostnamePortSpecification(INCLUDE_COMPLETE_HOST, EXCLUDE_COMPLETE_HOST)};
+        Domain d = new Domain("mydomain", "", Arrays.asList(ar));
+
+        testCreateUpdateDelete(credential, updatedCredential, d);
```

Comment:
I do not see the value in these tests. They just re-test core functionality in credentials plugin that cannot be broken by credentials impls. Adding useless tests is not a good idea imho